python/pyarrow/array.pyx

```diff
@@ -346,18 +347,21 @@ def from_pandas_series(object series, object mask=None, timestamps_to_ms=False):
         shared_ptr[CArray] out
 
     series_values = series_as_ndarray(series)
-    if series_values.dtype.type == np.datetime64 and timestamps_to_ms:
-        series_values = series_values.astype('datetime64[ms]')
+    utc_series = series.dt.tz_localize(None) if series_values.dtype.type == np.datetime64 else series
+
+    utc_series_values = series_as_ndarray(utc_series)
+    if utc_series_values.dtype.type == np.datetime64 and timestamps_to_ms:
+        utc_series_values = utc_series_values.astype('datetime64[ms]')
```

Comment:
To my knowledge, all pandas data having a time zone is UTC -- if data is time zone naive, then we may assume UTC in Arrow. I don't think any conversions are necessary here